webapp/src/main/java/org/apache/atlas/web/rest/TypeCacheRefreshREST.java

```diff
@@ -0,0 +1,94 @@
+package org.apache.atlas.web.rest;
+
+import org.apache.atlas.annotation.Timed;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.repository.RepositoryException;
+import org.apache.atlas.repository.graph.IAtlasGraphProvider;
+import org.apache.atlas.store.AtlasTypeDefStore;
+import org.apache.atlas.web.service.AtlasHealthStatus;
+import org.apache.atlas.web.service.ServiceState;
+import org.apache.atlas.web.util.Servlets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+
+import static org.apache.atlas.AtlasErrorCode.FAILED_TO_REFRESH_TYPE_DEF_CACHE;
+import static org.apache.atlas.repository.Constants.VERTEX_INDEX;
+
+
+@Path("admin/types")
+@Singleton
+@Service
+@Consumes({Servlets.JSON_MEDIA_TYPE, MediaType.APPLICATION_JSON})
+@Produces({Servlets.JSON_MEDIA_TYPE, MediaType.APPLICATION_JSON})
+public class TypeCacheRefreshREST {
+    private static final Logger LOG = LoggerFactory.getLogger(TypeCacheRefreshREST.class);
+
+    private final AtlasTypeDefStore typeDefStore;
+    private final IAtlasGraphProvider provider;
+    private final ServiceState serviceState;
+    private final AtlasHealthStatus atlasHealthStatus;
+
+    @Inject
+    public TypeCacheRefreshREST(AtlasTypeDefStore typeDefStore, IAtlasGraphProvider provider, ServiceState serviceState, AtlasHealthStatus atlasHealthStatus) {
+        this.typeDefStore = typeDefStore;
+        this.provider = provider;
+        this.serviceState = serviceState;
+        this.atlasHealthStatus = atlasHealthStatus;
+    }
+
+    /**
+     * API to refresh type-def cache.
+     *
+     * @throws AtlasBaseException
+     * @HTTP 204 if type def cache is refreshed successfully
+     * @HTTP 500 if there is an error refreshing type def cache
+     */
+    @POST
+    @Path("/refresh")
+    @Timed
+    public void refreshCache(@QueryParam("expectedFieldKeys") int expectedFieldKeys) throws AtlasBaseException, RepositoryException, InterruptedException {
+        try {
+            refreshTypeDef(expectedFieldKeys);
+        } catch (Exception e) {
+            LOG.error(e.getMessage(),e);
```

Comment:
Let's improve all the error logs by adding a custom message, so we can search on sentry.
Also our Rest APIsonly return AtlasBaseException -> convert other exceptions to this exception for consistency with a proper exception message

Reply:
This is addressed